### PR TITLE
APIs for PUT and GET Cluster Settings

### PR DESCRIPTION
**Pull request by abhivka7**
### Description
Creating Smithy APIs for PUT and GET Cluster Settings
 
### Issues Resolved
[#13](https://github.com/opensearch-project/opensearch-api-specification/issues/13) & [#14](https://github.com/opensearch-project/opensearch-api-specification/issues/14)

 
### Check List
- :white_check_mark: Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: model/cluster/get_cluster_settings/operations.smithy [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+namespace OpenSearch
+
+@externalDocumentation(
+    "OpenSearch Documentation": "https://opensearch.org/docs/latest/opensearch/rest-api/cluster-settings/"
+)
+
+@suppress(["HttpUriConflict"])
+@http(method: "GET", uri: "/_cluster/settings")
+@documentation("Returns cluster settings.")
+operation GETClusterSettings {
```
Comment (sachetalva) [inline model/cluster/get_cluster_settings/operations.smithy:16]: minor: can we change "GET" to "Get" in all operation / structure names?

File: model/common_datatypes.smithy [hunk 2]
```diff
@@ -18,3 +18,18 @@ map UserDefinedValueMap{
...
 
 document UserDefinedValue
+
+@enum([
+    {
+     value: "transient",
+     name: "transient"},
+    {
+     value: "persistent",
+     name: "persistent"}
```
Comment (sachetalva) [inline model/common_datatypes.smithy:28]: do we also need "defaults"?

File: model/cluster/get_cluster_settings/operations.smithy (hunk 1 above)
Comment (abhivka7) [reply model/cluster/get_cluster_settings/operations.smithy]: Done in new commit.

File: model/common_datatypes.smithy (hunk 2 above)
Comment (abhivka7) [reply model/common_datatypes.smithy]: Thanks for pointing this out Sachet. Done in latest commit.

**Review by pgtgrly — APPROVED** [review]

**Review by sachetalva — APPROVED** [review]

**Pull request merged**